### PR TITLE
feat(Navbar): Fixed Navbar Functionality

### PR DESCRIPTION
**Pull request by sandeep-v1404**
Signed-off-by: sandeep-v1404 <valpasanisandeep@gmail.com>

### Functionality:
Fix Bugs in Navbar

### Solution:
Modified Functions and Added New Function to make Navbar Functionality bug free.

### Risk level:
- [ ] high 
- [X] medium
- [ ] low

### How to test:

- ng serve
- firebase emulators:start
- Check Navbar Below Header.

### Screenshot:

- Login Page

image

- Start New Sprint

image

- Create New Task

image

- My Dashboard

image

- Tasks Page

image

- Task Details Page

image

### CODE REVIEW

**Review by bighnesh0404 — APPROVED** [review]

**Review by bighnesh0404 — APPROVED** [review]
Ship it

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ const routes: Routes = [
   { path: 'CreateNewSession', component: CreateNewSessionComponent },
   { path: 'StartNewSprint', component: CreateNewSprintComponent },
   { path: 'Tasks/:category/:currentSprintName', component: TasksComponent },
-  { path: 'TaskDetails/:taskId', component: TaskDetailsComponent },
+  { path: 'Tasks/:category/:currentSprintName/:taskId', component: TaskDetailsComponent },
```
Comment (viv9k) [inline src/app/app-routing.module.ts:21]: We don't need currentSprintName here

File: src/app/body/tasks/task-cards/task-cards.component.ts [hunk 2]
```diff
@@ -10,14 +10,15 @@ import { TasksId } from 'src/app/Interface/TasksInterface';
...
+  @Input("sprintName") sprintName: string
 
   constructor(private router: Router) { }
 
   ngOnInit(): void {
   }
 
   openTaskDetails(id: string) {
-    this.router.navigate(['/TaskDetails', id]);
+    this.router.navigate(['/Tasks', this.task.Category, this.sprintName, id]);
```
Comment (viv9k) [inline src/app/body/tasks/task-cards/task-cards.component.ts:21]: We don't need sprint name in route

File: src/app/body/tasks/tasks.component.html [hunk 3]
```diff
@@ -18,7 +18,7 @@ <h5 style="font-weight:bolder">Tasks</h5>
             <div class="col">
                 <div id="cardsList">
                     <ng-container *ngFor="let item of tasksData | async">
-                        <app-task-cards [task]="item"></app-task-cards>
+                        <app-task-cards [task]="item" [sprintName]="currentSprintName"></app-task-cards>
```
Comment (viv9k) [inline src/app/body/tasks/tasks.component.html:21]: We don't need to send current sprint name

File: src/app/body/tasks/tasks.component.ts [hunk 4]
```diff
@@ -29,15 +29,15 @@ export class TasksComponent implements OnInit {
...
+  constructor(private route: ActivatedRoute, private router: Router, private db: AngularFirestore, public navbarHandlerService: NavbarHandlerService) { }
 
   ngOnInit(): void {
 
     this.category = this.route.snapshot.params['category'];
     this.currentSprintName = this.route.snapshot.params['currentSprintName'];
 
-    this.componentName = this.category;
-    this.navbarHandler.addToNavbar(this.componentName);
+    this.navbarHandlerService.setNavbar("Board");
```
Comment (viv9k) [inline src/app/body/tasks/tasks.component.ts:39]: We don't need to set Board as Navbar here

File: src/app/services/navbar-handler.service.ts [hunk 5]
```diff
@@ -5,51 +5,24 @@ import { Injectable } from '@angular/core';
...
-      }
+    else {
+      this.navbarString += this.currentTab + " / "
+      this.currentTab = componentName1
     }
+  }
 
+  setNavbar(componentName: string) {
+    this.currentTab = componentName
+    this.navbarString = ""
```
Comment (viv9k) [inline src/app/services/navbar-handler.service.ts:26]: Please use old logic of navbar so that we can make use of it later

**Review by viv9k — CHANGES_REQUESTED** [review]
Please complete the required changes and get them completed.

File: src/environments/environment.ts [hunk 6]
```diff
@@ -5,7 +5,7 @@
 export const environment = {
   production: false,
   firebase: {
-    
+
```
Comment (viv9k) [inline src/environments/environment.ts:8]: revert this file

**Review by viv9k — CHANGES_REQUESTED** [review]
Fix the required changes asap.

File: src/environments/environment.ts (hunk 6 above)
Comment (sandeep-v1404) [reply src/environments/environment.ts]: Done Bhaiya

**Review by viv9k — APPROVED** [review]
ship it

**Comment by k0malSharma** [comment]
Please resolve merge conflicts as soon as possible

**Pull request merged**